cryostatio/cryostat-web
# feat(api): include sourceTarget details in download requests

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Related to https://github.com/cryostatio/cryostat/issues/760

## Description of the change:
This adds some form fields to API requests to generate JWT tokens for downloading recordings and reports.
These details are embedded in the request URL and are not so easy for the backend's abstract base class to figure out,
so the frontend provides them again as form parameters here.

## Motivation for the change:
The backend should verify that the provided form parameters actually match (are present within) the request URL, that
they are not blank, and that they correspond to actually known resources.

These details are useful for the backend to know so that it can perform further checks and actions when these JWT
download requests are made.

## How to manually test:
1. Check out https://github.com/cryostatio/cryostat/pull/1188
2. Update `web-client` submodule to the latest from this PR
3. `mvn package` to build a new container image with backend and frontend changes
4. Tag the image, push it to `quay.io`, and then deploy this in OpenShift using the Cryostat Operator
5. Open the Cryostat web UI as usual and click around
